### PR TITLE
implement Mail Sending after Customer Registration

### PR DESCRIPTION
**Pull request by gengjozsef**
- implement CustomerRegistrationMailCompanyUserPostCreatePlugin
- implement Mail Twig Templates
- implement Mail Type Plugin
- add autoloads in composer.json

### CODE REVIEW

**Review by sourcecube — APPROVED** [review]

File: bundles/company-user-mail-connector/src/FondOfOryx/Shared/CompanyUserMailConnector/Transfer/company_users_rest_api.transfer.xml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0"?>
+<transfers xmlns="spryker:transfer-01"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="spryker:transfer-01 http://static.spryker.com/transfer-01.xsd">
```
Comment (julianzimmermann) [inline bundles/company-user-mail-connector/src/FondOfOryx/Shared/CompanyUserMailConnector/Transfer/company_users_rest_api.transfer.xml:4]: https

File: bundles/company-user-mail-connector/src/FondOfOryx/Zed/CompanyUserMailConnector/Business/Model/Mail/MailHandler.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    public function sendMail(CompanyUserTransfer $companyUserTransfer): CompanyUserTransfer
+    {
+        $customerTransfer = $companyUserTransfer->getCustomer();
+
+        if ($customerTransfer->getIsNew() === false) {
+            return $companyUserTransfer;
+        }
+
+        $mailTransfer = new MailTransfer();
+        $mailTransfer->setType(CustomerRegisterationMailTypePlugin::MAIL_TYPE);
```
Comment (julianzimmermann) [inline bundles/company-user-mail-connector/src/FondOfOryx/Zed/CompanyUserMailConnector/Business/Model/Mail/MailHandler.php:39]: CustomerRegistrationMailTypePlugin

File: bundles/company-user-mail-connector/src/FondOfOryx/Zed/CompanyUserMailConnector/Business/Model/Mail/MailHandler.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace FondOfOryx\Zed\CompanyUserMailConnector\Business\Model\Mail;
+
+use FondOfOryx\Zed\CompanyUserMailConnector\Communication\Plugin\Mail\CustomerRegisterationMailTypePlugin;
```
Comment (julianzimmermann) [inline bundles/company-user-mail-connector/src/FondOfOryx/Zed/CompanyUserMailConnector/Business/Model/Mail/MailHandler.php:5]: CustomerRegistrationMailTypePlugin

**Review by julianzimmermann — APPROVED** [review]

File: bundles/company-user-mail-connector/src/FondOfOryx/Shared/CompanyUserMailConnector/Transfer/company_users_rest_api.transfer.xml (hunk 1 above)
Comment (gengjozsef) [reply bundles/company-user-mail-connector/src/FondOfOryx/Shared/CompanyUserMailConnector/Transfer/company_users_rest_api.transfer.xml:4]: @julianzimmermann Thanks. Resolved

File: bundles/company-user-mail-connector/src/FondOfOryx/Zed/CompanyUserMailConnector/Business/Model/Mail/MailHandler.php (hunk 2 above)
Comment (gengjozsef) [reply bundles/company-user-mail-connector/src/FondOfOryx/Zed/CompanyUserMailConnector/Business/Model/Mail/MailHandler.php]: @julianzimmermann Thanks. I Corrected it

File: bundles/company-user-mail-connector/src/FondOfOryx/Zed/CompanyUserMailConnector/Business/Model/Mail/MailHandler.php (hunk 3 above)
Comment (gengjozsef) [reply bundles/company-user-mail-connector/src/FondOfOryx/Zed/CompanyUserMailConnector/Business/Model/Mail/MailHandler.php]: @julianzimmermann Thanks. I've Corrected it

**Pull request merged**